docs/sources/setup-grafana/configure-grafana/_index.md

```diff
@@ -858,6 +860,21 @@ By default, the users' organization and role is reset on every new login.
 > With Grafana 10, if `oauth_skip_org_role_update_sync` option is set to `false`, users with no mapping will be
 > reset to the default organization role on every login. [See `auto_assign_org_role` option]({{< relref ".#auto_assign_org_role" >}}).
 
+### [auth].<provider> skip_org_role_update_sync
+
+If you want to prevent org roles from being synced for a specific OAuth integration, set `skip_org_role_update_sync` to `true`.
+
+The `oauth_skip_org_role_update_sync` are independent of each other. `skip_org_role_sync` is specific to the provider integration, while `oauth_skip_org_role_sync_update` is a generic setting that only applies to `Github`, `Gitlab`, `AzureAD`, `Generic OAuth` integrations. We are deprecating the generic setting in favor of the provider specific settings.
+
+A table detailing what happens when the legacy option is enabled/disabled and an auth specific version is enabled/disabled is shown below.
+
+The table below shows the available OAuth providers and their setting with the default value and the skip org role sync setting.
+| OAuth Provider | `oauth_skip_org_role_sync_update` | `skip_org_role_sync` | Behavior |
+| --- | --- | --- | --- | --- |
+| Grafana.com | true | false | will still sync Org Roles from Grafana.Com |
```

Comment:
```suggestion
| Grafana.com | true | false | skip org role sync for grafana.com |
```

This should be the correct behavior otherwise it will break existing user's configs

Reply:
We do not have any Grafana.com for the "generic" setting. Meaning that the existing behavior is syncing the org roles for GrafanaCom, this is always why this is being prioritized.

If we want to change existing behavior for users then we will have to introduce GrafanaCom in the sync part as well.

---
Summarize:
- Do we want eventually that all of the OAuth integrations will skip the sync if the "generic" setting is set from the users; then we have to make this change and it will affect the behavior for current users.

Reply:
Maybe the frontend is not taking it into account but the backend is

https://github.com/grafana/grafana/blob/main/pkg/api/login_oauth.go#L288 should affect all oauth integrations

 We had to change this setting for ops.grafana because we enforced the sync for a while.

Reply:
I don't really follow your comment Eric, but I think what Joao suggested is the behaviour that we want and is also the behaviour that we have after the changes in this PR - org role sync for GCom will be disabled if either the generic `oauth_skip_org_role_sync_update` is set to true or if `skip_org_role_sync` for GCom is set to true.

This is where the generic setting is looked at when setting org roles for the external user:
https://github.com/grafana/grafana/blob/main/pkg/api/login_oauth.go#L288

Note that I haven't tested this myself, but I think this is the behaviour that we saw when we were testing together Eric?

Reply:
before I made my comment I tested this out, let me get back to you with a test to see that was true. Yeas to me I login and the setting `oauth_skip_org_role_sync_update` is set to `true`.


https://user-images.githubusercontent.com/7727602/211037777-3420ccd6-c72e-426d-9340-d7ea3a6f88ff.mov

Does this make sense to you? That we are unable to set the org roles "basic,viewer and admin"

Reply:
when we have the other setting enabled we are able to edit the org roles.

here the setting `skip_org_role_sync` in `[auth].grafanacom` is set to `true`.

https://user-images.githubusercontent.com/7727602/211038725-c2a5f15c-5977-4aaf-998f-7f9e6304b851.mov

Reply:
if we want it to be that `oauth_skip_org_role_sync_update` is set to `true` would enable us to also edit the org roles. Then we would need to change this PR to reflect that

Reply:
hey! i am realizing i am only checking the frontend settings here and we are talking about what the function signature for the externalUser is sending. Let me revisit this

Reply:
> Maybe the frontend is not taking it into account but the backend is

summary:
- backend is taking it into account but not the frontend

changing the frontend to make it reflect what want as desired behaviour

Reply:
:heavy_check_mark:  changing the frontend to make it reflect what want as desired behaviour

:heavy_check_mark:  desired behavior being - we want the "generic" setting to make it so that all OAuth integrations are not synced when set. And when a specific provider is set, only those users should not be synced